### ISSUE TITLE
Feature Request - LabelGeometryImageFilter

### ISSUE THREAD

**Issue by kvrd18** [SimpleITK/SimpleITK#312]
Hi,

Is `LabelGeometryImageFilter` in the pipeline for future releases? If not, can you please add it?

**Comment by blowekamp** [comment]
Have you looked at the `LabelShapeStatisticsImageFilter`? This has most of the same functionality and is in a robust and tested framework. The current SimpleITK master as added computation for oriented bounding boxes too. Is there something missing with the filter?


The `LabelGeometryImageFilter` is only the itk Review module. I have encountered several issues with the filter with supporting spacing, and direction cosine matrix, along with computational efficiency.

**Comment by kvrd18** [comment]
Which function in SimpleITK have `GetMajorAxisLength` and `GetMinorAxisLength` from ITK's original `LabelGeometryImageFilter`?

**Comment by blowekamp** [comment]
Are you working in 2d or 3d? Do you want the extrema for size or a more robust measurement of size?

**Comment by kvrd18** [comment]
I'm working in 3D. For a given label, I need the extrema for size.

**Comment by blowekamp** [comment]
I just merged a [topic](#189) into SimpleITK's next branch which adds the new OrientedBoundingBox attributes to the `LabelShapeStatisticsImageFilter`. This branch will be merged into the master branch shortly after complete testing. To use this feature you will need to compile SimpleITK from the latest next branch.

The LabelGeometryImageFilter's `AxisLength` correspond to the `OrientedBoundingBoxSize`. The oriented bounding box is derived from the principle axis computation, then looking for the extrema in each direction. There have been recent improvement to ITK to address this computation to work correctly with direction cosine matrices. More information on the label map frame work can be obtainer in the original [Insight Journal paper](https://hdl.handle.net/1926/584).

Here is sample usage:
```

In [3]: l = sitk.ReadImage("Testing/Data/Input/2th_cthead1.png")

In [4]: lss = sitk.LabelShapeStatisticsImageFilter()

In [5]: lss.ComputeOrientedBoundingBoxOn()

In [6]: lss.Execute(l)

In [7]: lss.GetOrientedBoundingBoxSize(100)
Out[7]: (226.82865897088314, 293.59130481172144)

In [8]: lss.GetOrientedBoundingBoxSize(100)[-1]
Out[8]: 293.59130481172144

```

**Comment by blowekamp** [comment]
With the OBB topic now merged into master. This add the requested[ GetOrientedBoundingBoxSize](https://itk.org/SimpleITKDoxygen/html/classitk_1_1simple_1_1LabelShapeStatisticsImageFilter.html#a37215b476cffdb78c11e428dcc67ebd5) method to the interface. I am going to close this issue. Please feel free to reopen this topic if needed to continue the discussion.
